### PR TITLE
(FACT-1510) Use absolute rather than canonical

### PR DESCRIPTION
**Pull request by MikaelSmith**
Similar to FACT-1276, it's not necessary to get the full canonical path
for external facts dir - we only need it to be absolute. Attempting to
canonicalize paths is more expensive, and can fail on Windows with
symlinks.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
I'm a little concerned about what logging we lose from this.

**Comment by peterhuene** [comment]
👍I think we can do without that logged message; seems like the only condition when `absolute` throws is when memory allocation fails.

**Comment by MikaelSmith** [comment]
Before it'd warn if the directory didn't exist. Unfortunately sometimes it does that incorrectly.

**Comment by MikaelSmith** [comment]
Also, still verifying that this fixes the problem.

**Comment by MikaelSmith** [comment]
Retargeted at LTS.

File: lib/src/ruby/module.cc [hunk 1]
```diff
@@ -297,12 +297,8 @@ namespace facter { namespace ruby {
         for (auto dir : paths) {
             _additional_search_paths.emplace_back(dir);
 
-            // Get the canonical directory name
-            boost::system::error_code ec;
-            path directory = canonical(_additional_search_paths.back(), ec);
-            if (ec) {
-                continue;
-            }
+            // Get the absolute directory name
```
Comment (ferventcoder) [inline lib/src/ruby/module.cc]: Any chance you can add more comments as to why absolute over canonical (in that canonical causes errors with symlinks when the targets are not the same file system type (likely due to stat)

Comment (ferventcoder) [inline lib/src/ruby/module.cc]: I just don't want someone else to come along later and not research why this change was made and go back to canonical.

**Comment by ferventcoder** [comment]
> Similar to FACT-1276, it's not necessary to get the full canonical path
for external facts dir - we only need it to be absolute. Attempting to
canonicalize paths is more expensive, and can fail on Windows with
symlinks.

As far as I know, it only fails on Windows when attempting to use a symlink that is not also NTFS. This is likely due to things like stat or checking permissions. We had similar issues in Puppet::FileSystem that needed to change.

File: lib/src/ruby/module.cc (hunk 1 above)
Comment (MikaelSmith) [inline lib/src/ruby/module.cc]: In comments, rather than the commit? I guess I can add it to the first instance.

**Comment by MikaelSmith** [comment]
Updated.

File: lib/src/ruby/module.cc (hunk 1 above)
Comment (ferventcoder) [inline lib/src/ruby/module.cc]: In commit, but also in comments 👍

**Comment by MikaelSmith** [comment]
Ping @ferventcoder, @puppetlabs/facter-maintainers

**Comment by ferventcoder** [comment]
LGTM - thanks for adding those comments!

**Comment by Magisus** [comment]
👍

**Pull request merged**